contracts/optimistic-ethereum/OVM/chain/OVM_ChainStorageContainer.sol

```diff
@@ -0,0 +1,222 @@
+pragma solidity ^0.7.0;
+
+/* Library Imports */
+import { Lib_RingBuffer } from "../../libraries/utils/Lib_RingBuffer.sol";
+import { Lib_AddressResolver } from "../../libraries/resolver/Lib_AddressResolver.sol";
+
+/* Interface Imports */
+import { iOVM_ChainStorageContainer } from "../../iOVM/chain/iOVM_ChainStorageContainer.sol";
+
+/**
+ * @title OVM_ChainStorageContainer
+ */
+contract OVM_ChainStorageContainer is iOVM_ChainStorageContainer, Lib_AddressResolver {
+
+    /*************
+     * Libraries *
+     *************/
+
+    using Lib_RingBuffer for Lib_RingBuffer.RingBuffer;
+
+
+    /*************
+     * Variables *
+     *************/
+
+    string public owner;
+    Lib_RingBuffer.RingBuffer internal buffer;
+
+
+    /***************
+     * Constructor *
+     ***************/
+    
+    /**
+     * @param _libAddressManager Address of the Address Manager.
+     * @param _owner Name of the contract that owns this container (will be resolved later).
+     */
+    constructor(
+        address _libAddressManager,
+        string memory _owner
```

Comment:
Makes the library authenticated to a particular contract (resolved via `AddressManager`).